doc/architecture-decisions/0018-route-crd-and-contour-controller.md

```diff
@@ -0,0 +1,48 @@
+# 18. Add Contour as an Alternative Ingress Solution Provider
+
+Date: 2020-10-09
+
+## Status
+
+Proposal
+
+## Context
+In our efforts to allow more optionality in ingress solutions for CF-for-K8s, we
+want to allow the use of Contour as a potential alternative to Istio.
+
+### Proposed Design
+
+We propose:
+1. Adding a configuration option `ingress_solution_provider` to CF-for-K8s, with
+   potential values `istio` or `contour`.
+2. Creating another kubebuilder controller, similar to routecontroller, to
+   handle when `ingress_solution_provider: contour` is configured.
+3. Renaming `Route Controller` to `Istio Controller`
+
+![Proposed ContourControllerDesign](../assets/contour-controller.png)
+
+### Open Questions
+1. What do we do about the config? Does contour config live in cf-k8s-networking
+   and eventually it moves to cf-for-k8s?
+2. What happens if an operator wants to change their ingress solution provider,
+   will they have to redeploy CF-for-K8s? Is that a big deal?
+3. Looks like Contour deploys Envoys as a daemonset. Is that gonna be a problem?
```

Comment:
I think they do that in the quick start for the same reason we do/did. With LoadBalancer services it's not required, so we'll probably just have a recommended way to change it to a DaemonSet for bigger deployments.

Reply:
@XanderStrike I am assuming you meant changing Envoy to a "Deployment" for big deployments, is so that is true and we can apply some `ytt` magic [here](https://github.com/projectcontour/contour/blob/main/examples/contour/03-envoy.yaml#L3).

Reply:
Yes exactly, thanks